File: LeetCode/2315. Count Asterisks/solution.py
"""
Runtime: 44 ms
Memory Usage: 15.1 MB
"""
class Solution:
    def countAsterisks(self, s: str) -> int:
        open = True
        ans = 0
        for c in s:
            if c == '|':
                open = not open
            if open and c == '*':
                ans += 1
        return ans

File: LeetCode/2315. Count Asterisks/solution.cpp
/**
 * Runtime: 4 ms
 * Memory Usage: 6.3 MB 
 */
class Solution {
public:
    int countAsterisks(string s) {
        bool open = true;
        int ans = 0;
        for (auto &c: s) {
            if (c == '|') {
                open = !open;
            }
            if (open && c == '*') {
                ++ ans;
            }
        }
        return ans;
    }
};
